### PR TITLE
Add missing documentation in EnvelopedCms and update existing docs.

### PR DESCRIPTION
**Pull request by bartonjs**
* Adds documentation to members with no documentation.
* Updates existing documentation to match style guidelines.
* Clarifies and corrects some information.

cc: @carlossanlop @mairaw @rpetrusha

### CODE REVIEW

File: xml/System.Security.Cryptography.Pkcs/EnvelopedCms.xml [hunk 1]
```diff
@@ -278,9 +288,17 @@
...
+        <remarks>
+          <format type="text/markdown"><![CDATA[  
+  
+## Remarks  
+ When the <xref:System.Security.Cryptography.Pkcs.EnvelopedCms> class is being used to to encrypt data, the contents of this collection are embedded into the pending message during the call to one of the <xref:System.Security.Cryptography.Pkcs.EnvelopedCms.Encrypt%2A> methods. Changes after the encryption operation have no effect.
+
+ The value of this property is replaced with a different collection object during a call to <xref:System.Security.Cryptography.Pkcs.EnvelopedCms.Decode%2A>, which then represents the certificates that were read out of the decoded message.
+  
+ ]]></format>
       </Docs>
```
Comment (bartonjs) [inline xml/System.Security.Cryptography.Pkcs/EnvelopedCms.xml]:
```suggestion
        </remarks>
      </Docs>
```

File: xml/System.Security.Cryptography.Pkcs/EnvelopedCms.xml [hunk 2]
```diff
@@ -198,20 +208,20 @@
...
-        <param name="contentInfo">A <see cref="T:System.Security.Cryptography.Pkcs.ContentInfo" /> object that represents the content and its type.</param>
-        <summary>The <see cref="M:System.Security.Cryptography.Pkcs.EnvelopedCms.#ctor(System.Security.Cryptography.Pkcs.SubjectIdentifierType,System.Security.Cryptography.Pkcs.ContentInfo)" /> constructor creates an instance of the <see cref="T:System.Security.Cryptography.Pkcs.EnvelopedCms" /> class by using the specified subject identifier type and content information. The specified content information is to be used as the inner content type.</summary>
+        <param name="recipientIdentifierType">One of the enumeration values that specifies the subject identifier scheme to use for recipient certificates selected by a UI prompt.</param>
+        <param name="contentInfo">The message content to encrypt.</param>
+        <summary>Initializes a new instance of the <see cref="T:System.Security.Cryptography.Pkcs.EnvelopedCms" /> class with specified content information, and a subject identifier scheme when selecting recipients by UI.</summary>
```
Comment (rpetrusha) [inline xml/System.Security.Cryptography.Pkcs/EnvelopedCms.xml]:
```suggestion
        <summary>Initializes a new instance of the <see cref="T:System.Security.Cryptography.Pkcs.EnvelopedCms" /> class with specified content information and a subject identifier scheme when selecting recipients by UI.</summary>
```

File: xml/System.Security.Cryptography.Pkcs/EnvelopedCms.xml [hunk 3]
```diff
@@ -198,20 +208,20 @@
...
         <remarks>
           <format type="text/markdown"><![CDATA[  
   
 ## Remarks  
- For the default property values used with this constructor, see <xref:System.Security.Cryptography.Pkcs.EnvelopedCms.%23ctor>.  
-  
- Subjects can be identified by either the issuer and serial number of their public key certificate or by the key identifier of their public key certificate.  
+ On .NET Framework the default symmetric encryption algorithm is determined by the version of the .NET Framework that the application executable was built against. Starting with .NET Framework 4.8 the default algorithm is AES-256 in CBC mode, in prior versions the default is TripleDES (DES3-EDE) in CBC mode.
```
Comment (rpetrusha) [inline xml/System.Security.Cryptography.Pkcs/EnvelopedCms.xml]:
```suggestion
 On .NET Framework, the default symmetric encryption algorithm is determined by the version of the .NET Framework that the application executable was built against. Starting with .NET Framework 4.8, the default algorithm is AES-256 in CBC mode; in prior versions, the default is TripleDES (DES3-EDE) in CBC mode.
```

File: xml/System.Security.Cryptography.Pkcs/EnvelopedCms.xml [hunk 4]
```diff
@@ -198,20 +208,20 @@
...
+
+ On .NET Core the default symmetric encryption algorithm is determined by the version of the System.Security.Cryptography.Pkcs assembly which gets loaded. Starting with the 4.6.0 version of the System.Security.Cryptography.Pkcs NuGet package the default algorithm is AES-256 in CBC mode, in prior versions the default is TripleDES (DES3-EDE) in CBC mode.
```
Comment (rpetrusha) [inline xml/System.Security.Cryptography.Pkcs/EnvelopedCms.xml]:
```suggestion
 On .NET Core, the default symmetric encryption algorithm is determined by the version of the System.Security.Cryptography.Pkcs assembly that is loaded. Starting with the 4.6.0 version of the System.Security.Cryptography.Pkcs NuGet package, the default algorithm is AES-256 in CBC mode; in prior versions, the default is TripleDES (DES3-EDE) in CBC mode.
```

File: xml/System.Security.Cryptography.Pkcs/EnvelopedCms.xml [hunk 5]
```diff
@@ -349,13 +370,15 @@
         <ReturnType>System.Security.Cryptography.Pkcs.ContentInfo</ReturnType>
       </ReturnValue>
       <Docs>
-        <summary>The <see cref="P:System.Security.Cryptography.Pkcs.EnvelopedCms.ContentInfo" /> property retrieves the inner content information for the enveloped CMS/PKCS #7 message.</summary>
-        <value>A <see cref="T:System.Security.Cryptography.Pkcs.ContentInfo" /> object that represents the inner content information from the enveloped CMS/PKCS #7 message.</value>
+        <summary>Gets the content information for the enveloped CMS/PKCS#7 message.</summary>
+        <value>Gets the content information for the enveloped CMS/PKCS#7 message.</value>
```
Comment (rpetrusha) [inline xml/System.Security.Cryptography.Pkcs/EnvelopedCms.xml]:
```suggestion
        <value>The content information for the enveloped CMS/PKCS#7 message.</value>
```

File: xml/System.Security.Cryptography.Pkcs/EnvelopedCms.xml [hunk 6]
```diff
@@ -349,13 +370,15 @@
...
         <remarks>
           <format type="text/markdown"><![CDATA[  
   
 ## Remarks  
- If you call any of the decrypt methods, such as <xref:System.Security.Cryptography.Pkcs.EnvelopedCms.Decrypt>, before viewing this property, the content returned is encrypted.  
+ When the <xref:System.Security.Cryptography.Pkcs.EnvelopedCms> class is being used to to encrypt data, the value of this property represents the data to encrypt. It is not replaced with encrypted content after a call to an <xref:System.Security.Cryptography.Pkcs.EnvelopedCms.Encrypt%2A> method.
+
+ The value of this property is replaced during a call to <xref:System.Security.Cryptography.Pkcs.EnvelopedCms.Decode%2A>, which then represents the encrypted content decoded from the message. After a successful call to a <xref:System.Security.Cryptography.Pkcs.EnvelopedCms.Decrypt%2A> method the value is replaced again, with the decrypted message contents.
```
Comment (rpetrusha) [inline xml/System.Security.Cryptography.Pkcs/EnvelopedCms.xml]:
```suggestion
 The value of this property is replaced during a call to <xref:System.Security.Cryptography.Pkcs.EnvelopedCms.Decode%2A>, which then represents the encrypted content decoded from the message. After a successful call to a <xref:System.Security.Cryptography.Pkcs.EnvelopedCms.Decrypt%2A> method, the value is replaced again, with the decrypted message contents.
```

File: xml/System.Security.Cryptography.Pkcs/EnvelopedCms.xml [hunk 7]
```diff
@@ -594,10 +611,21 @@
...
-        <summary>To be added.</summary>
-        <remarks>To be added.</remarks>
+        <param name="recipientInfo">The recipient info to use for decryption.</param>
+        <param name="privateKey">The private key to use to decrypt the recipient-specific information.</param>
+        <summary>Decrypts the contents of the decoded enveloped CMS/PKCS#7 message via a specified recipient info with a specified private key.</summary>
+        <remarks>
+          <format type="text/markdown"><![CDATA[  
+  
+## Remarks  
+Unlike the other overloads of Decrypt, this overload does not search certificate stores. The <xref:System.Security.Cryptography.Pkcs.SubjectIdentifier.MatchesCertificate(System.Security.Cryptography.X509Certificates.X509Certificate2)> method can be used to manually match an expected certificate to the <xref:System.Security.Cryptography.Pkcs.RecipientInfo.RecipientIdentifier?displayProperty=nameWithType> value to evaluate the applicability of the private key to the recipient info value.
```
Comment (rpetrusha) [inline xml/System.Security.Cryptography.Pkcs/EnvelopedCms.xml]:
```suggestion
Unlike the other overloads of `Decrypt`, this overload does not search certificate stores. The <xref:System.Security.Cryptography.Pkcs.SubjectIdentifier.MatchesCertificate(System.Security.Cryptography.X509Certificates.X509Certificate2)> method can be used to manually match an expected certificate to the <xref:System.Security.Cryptography.Pkcs.RecipientInfo.RecipientIdentifier?displayProperty=nameWithType> value to evaluate the applicability of the private key to the recipient info value.
```

File: xml/System.Security.Cryptography.Pkcs/EnvelopedCms.xml [hunk 8]
```diff
@@ -735,16 +759,14 @@
       </ReturnValue>
       <Parameters />
       <Docs>
-        <summary>The <see cref="M:System.Security.Cryptography.Pkcs.EnvelopedCms.Encrypt" /> method encrypts the contents of the CMS/PKCS #7 message.</summary>
+        <summary>Encrypt the contents of the CMS/PKCS#7 message after selecting recipient certificates via a UI popup.</summary>
```
Comment (rpetrusha) [inline xml/System.Security.Cryptography.Pkcs/EnvelopedCms.xml]:
```suggestion
        <summary>Encrypts the contents of the CMS/PKCS#7 message after selecting recipient certificates via a UI popup.</summary>
```

File: xml/System.Security.Cryptography.Pkcs/EnvelopedCms.xml [hunk 9]
```diff
@@ -786,18 +808,12 @@
...
-          <format type="text/markdown"><![CDATA[  
-  
-## Remarks  
- The <xref:System.Security.Cryptography.Pkcs.EnvelopedCms.Encrypt%28System.Security.Cryptography.Pkcs.CmsRecipient%29> method does not cause the <xref:System.Security.Cryptography.Pkcs.EnvelopedCms.RecipientInfos%2A?displayProperty=nameWithType> property to be populated.  
-  
- ]]></format>
-        </remarks>
-        <exception cref="T:System.ArgumentNullException">A null reference was passed to a method that does not accept it as a valid argument.</exception>
+        <param name="recipient">The recipient information describing the single recipient of this message.</param>
+        <summary>Encrypts the contents of the CMS/PKCS#7 message for a single, specified recipient.</summary>
```
Comment (rpetrusha) [inline xml/System.Security.Cryptography.Pkcs/EnvelopedCms.xml]:
```suggestion
        <summary>Encrypts the contents of the CMS/PKCS#7 message for a single specified recipient.</summary>
```

**Comment by carlossanlop** [comment]
@rpetrusha @mairaw the suggestions where applied and the build passed. Is this good to merge?

**Comment by rpetrusha** [comment]
Yes, this is good to merge, @carlossanlop and @bartonjs. I'll merge it now.

**Pull request merged**